test/e2e/case1_audit_from_cache_test.go

```diff
@@ -0,0 +1,334 @@
+
+				return config.Spec.Sync.SyncOnly
+			}, timeout).Should(HaveLen(3))
+		})
+		It("Should have 2 syncOnly elements in config", func() {
+			Kubectl("delete", "-f", case1ConstraintIngressYaml, "--ignore-not-found")
+
+			config := &v1alpha1.Config{}
+
+			Eventually(func(g Gomega) []v1alpha1.SyncOnlyEntry {
+				err := K8sClient.Get(ctx, types.NamespacedName{Name: "config", Namespace: gatekeeperNamespace}, config)
+				g.Expect(err).ShouldNot(HaveOccurred())
+
+				return config.Spec.Sync.SyncOnly
+			}, timeout).Should(HaveLen(2))
+
+			By("Ingress should not be in SyncOnly")
+			Expect(slices.IndexFunc(config.Spec.Sync.SyncOnly, func(s v1alpha1.SyncOnlyEntry) bool {
+				return s.Kind == "Ingress"
+			})).Should(Equal(-1))
+		})
+		It("Should have 1 syncOnly elements in config", func() {
+			Kubectl("delete", "-f", case1ConstraintStorageclassYaml, "--ignore-not-found")
+
+			config := &v1alpha1.Config{}
+
+			Eventually(func(g Gomega) []v1alpha1.SyncOnlyEntry {
+				err := K8sClient.Get(ctx, types.NamespacedName{Name: "config", Namespace: gatekeeperNamespace}, config)
+				g.Expect(err).ShouldNot(HaveOccurred())
+
+				return config.Spec.Sync.SyncOnly
+			}, timeout).Should(HaveLen(1))
+
+			By("StorageClass should not be in SyncOnly")
+			Expect(slices.IndexFunc(config.Spec.Sync.SyncOnly, func(s v1alpha1.SyncOnlyEntry) bool {
+				return s.Kind == "StorageClass"
+			})).Should(Equal(-1))
+		})
+		It("Should still have 1 syncOnly elements in config when Pod constraint is deleted", func() {
+			Kubectl("delete", "-f", case1ConstraintPodYaml, "--ignore-not-found")
+			config := &v1alpha1.Config{}
+
+			Eventually(func(g Gomega) []v1alpha1.SyncOnlyEntry {
+				err := K8sClient.Get(ctx, types.NamespacedName{Name: "config", Namespace: gatekeeperNamespace}, config)
+				g.Expect(err).ShouldNot(HaveOccurred())
+
+				return config.Spec.Sync.SyncOnly
+			}, timeout).Should(HaveLen(1))
+
+			By("Pod should exist in SyncOnly because case1ConstraintPod2 yet exist")
+			Expect(slices.IndexFunc(config.Spec.Sync.SyncOnly, func(s v1alpha1.SyncOnlyEntry) bool {
+				return s.Kind == "Pod"
+			})).ShouldNot(Equal(-1))
+		})
+		It("Should have 0 syncOnly elements in config ", func() {
+			Kubectl("delete", "-f", case1ConstraintPod2Yaml, "--ignore-not-found")
+			config := &v1alpha1.Config{}
+
+			Eventually(func(g Gomega) []v1alpha1.SyncOnlyEntry {
+				err := K8sClient.Get(ctx, types.NamespacedName{Name: "config", Namespace: gatekeeperNamespace}, config)
+				g.Expect(err).ShouldNot(HaveOccurred())
+
+				return config.Spec.Sync.SyncOnly
+			}, timeout).Should(BeNil())
+		})
+	})
+	Describe("Updating contraint should apply to config resource", Ordered, func() {
+		It("Should update the config resource", func() {
+			config := &v1alpha1.Config{}
+
+			By("Add a new constraint")
+			Kubectl("apply", "-f", case1ConstraintUpdateYaml)
+
+			By("Group name 'apps.StatefulSet' should exist in SyncOnly although the crds does not exist")
```

Comment:
Is this description accurate after recent changes?

Reply:
oh need to change description